maestro-orchestra/src/main/java/maestro/orchestra/workspace/MediaCommandsValidator.kt

```diff
@@ -0,0 +1,53 @@
+package maestro.orchestra.workspace
+
+import maestro.orchestra.error.MediaFileNotFound
+import maestro.orchestra.yaml.YamlCommandReader
+import okio.Path.Companion.toPath
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.notExists
+
+class MediaCommandsValidator(private val input: Path) {
```

Comment:
Yes, I realized that while testing. I have changed that accordingly.

Reply:
Shouldn't that also be the same logic for all path references in the flow, not just media files?

Reply:
Yes unifying this logic and removing this means the same nature as run flow:

1. Running locally  (both `maestro test single_flow.yaml` and `maestro test ./workspace`) would fail in one case when the path doesn't exist
2. With maestro cloud command it fail in both cases if its missing in workspace or not existing.

Reply:
I meant that this should also be the same logic for subflow paths:

> Yes, I realized that while testing. I have changed that accordingly. Now this file only checks if all the media files are in workspace in case of input being workspace directory.

So in that case, I'm curious why this validation logic is written in a class specific to media

Reply:
* If path doesn't exist: Should fail in all cases
* If path is outside the workspace: Should fail for `maestro test workspace/` and `maestro cloud workspace/`
* `maestro test single_flow.yaml` doesn't have a concept of a workspace yet so it should only check for existence

Reply:
> If path is outside the workspace: Should fail for maestro test workspace/ and maestro cloud workspace/

For this case I tried with runFlow command and it doesn't fail in case of `maestro test workspace` command. Is that intended?

Reply:
I don't think that was intentional. To me it would make sense for this case to be an error

Reply:
Hey @Leland-Takamine I fixed this by creating a common implementation which I am planning to raise as a separate PR. Can you have look at the latest commit if the abstraction looks good and then I'll merge this and raise the runFlow command PR. It would basically be 

```
allFlows.forEach {
            val commands = YamlCommandReader.readCommands(it).mapNotNull {  it.runFlowCommand }
            val flowPaths = commands.flatMap { it.sourceDescription }
            YamlCommandsPathValidator.validatePathsExistInWorkspace(input, it, flowPaths)
}
```

Reply:
Sounds good thanks!